janus_core/calculations/md.py

```diff
@@ -423,14 +423,20 @@ def __init__(
                 "heating to run",
                 stacklevel=2,
             )
-        if self.ramp_temp and self.ensemble in ("nve", "nph"):
-            raise ValueError(
-                "Temperature ramp requested for ensemble with no thermostat."
-            )
+        if self.ramp_temp:
+            if self.ensemble in ("nve", "nph"):
+                raise ValueError(
+                    "Temperature ramp requested for ensemble with no thermostat"
+                )
 
-        # Check validate start and end temperatures
-        if self.ramp_temp and (self.temp_start < 0 or self.temp_end < 0):
-            raise ValueError("Start and end temperatures must be positive")
+            # Check validate start and end temperatures
+            if self.temp_start < 0 or self.temp_end < 0:
+                raise ValueError("Start and end temperatures must be positive")
+
+            if self.temp_time < self.timestep:
+                raise ValueError(
+                    "Temperature ramp step time cannot be less than 1 timestep"
+                )
```

Comment:
I think when we discussed this, @oerc0122 was suggesting it might be better to warn, rather than fail.

My concern with that is, most people probably won't see the warning (which is captured by the log).

I think it really depends whether there are any circumstances where just having step 0 for each temperature may be useful.